Task_1_4_1/src/main/java/RecordBook.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.OptionalDouble;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class RecordBook {
+
+    private final List<Semester> semesterList = new ArrayList<>();
+    private Subject diplomaWork;
+
+    /**
+     * Добавить семестр с оценками.
+     *
+     * @param semester добавляемый семестр.
+     */
+    public void addSemester(Semester semester) {
+        semesterList.add(semester);
+    }
+
+    /**
+     * Получить объект последнего семестра.
+     *
+     * @return объект последнего семестра, в котором можно изменять оценки.
+     */
+    public Semester getLastSemester() {
+        if (semesterList.isEmpty()) {
+            return null;
+        }
+
+        return semesterList.get(semesterList.size() - 1);
+    }
+
+    /**
+     * Установить оценку для дипломной работы.
+     *
+     * @param subject дипломная работа.
+     */
+    public void setDiplomaWork(Subject subject) {
+        diplomaWork = subject;
+    }
+
+    private Stream<Subject> getAllSubjectsStream() {
+        Stream<Subject> subjectStream = Stream.empty();
+        for (var semester : semesterList) {
+            subjectStream = Stream.concat(subjectStream, semester.getSubjectList().stream());
+        }
+        return subjectStream;
```

Comment:
Это можно сделать гораздо проще с использованием flatMap
```suggestion
     return semesterList.stream()
            .flatMap(semester -> semester.getSubjectList().stream());
```